src/Soil-Core/SoilIndexedDictionary.class.st

```diff
@@ -339,7 +339,12 @@ SoilIndexedDictionary >> second [
 { #category : #accessing }
 SoilIndexedDictionary >> size [ 
 	^ transaction 
-		ifNotNil: [ self index size ]
+		ifNotNil: [ 
+			"We should just call size on index, but the do: here can restore values"
+			| sum |
+			sum := 0.
+			self do: [ :each | sum := sum + 1 ].
```

Comment:
this does not scale at all. Not sure how we do it until now but doing a linear scan counting is not the way to go

Reply:
But we do already do the same linear scam on the index...

Reply:
```
SoilBasicSkipList>>#size
	"We iterate over all elements to get the size. Slow!"
	^ self newIterator size 
```